lib/postal/smtp_server/server.rb

```diff
@@ -221,23 +233,23 @@ def run_event_loop
                   logger.error "[#{client_id}] #{line}"
                 end
                 # Close all IO and forget this client
-                @io_selector.deregister(io) rescue nil
-                buffers.delete(io)
-                io.close rescue nil
-                if @io_selector.empty?
-                  Process.exit(0)
-                end
+                # @io_selector.deregister(io) rescue nil
+                buffers.delete(connection)
+                connection.close rescue nil
+                break
+                # if @io_selector.empty?
+                #   Process.exit(0)
+                # end
               end
             end
           end
+          # end
           # If unlisten has been called, stop listening
           if $unlisten
-            @io_selector.deregister(@server)
+            # @io_selector.deregister(@server)
             @server.close
             # If there's nothing left to do, shut down the process
-            if @io_selector.empty?
-              Process.exit(0)
-            end
+            Process.exit(0) if @workers_pool.empty?
```

Comment:
Can this condition ever happen? I don't see anything that tries to shutdown the threads once the unlisten method is called.

Reply:
The condition replaces the previous condition.

Reply:
Sure. But I don't see anything that actually tries to empty the workers_pool, there's nothing I can see that shuts it down.

Reply:
You are right. It needs to be changed since I move to concurrent-ruby gem. I missed it. With concurrent-ruby it should be something like
```
Process.exit(0) if @workers_pool.queue_length == 0
```

Reply:
Fixed it